guard/tests/validate.rs

```diff
@@ -328,44 +365,366 @@ mod validate_tests {
+    )]
+    #[case(
+        vec!["db_resource.yaml"],
+        vec!["db_param_port_rule.guard"],
+        vec!["input-parameters-dir/db_params.yaml", "input-parameters-dir/db_metadata.yaml"],
+        StatusCode::SUCCESS
+    )]
+    #[case(
+        vec!["db_resource.yaml"],
+        vec!["db_param_port_rule.guard"],
+        vec!["input-parameters-dir/"],
+        StatusCode::SUCCESS
+    )]
+    #[case(
+        vec!["db_resource.yaml"],
+        vec!["db_param_port_rule.guard"],
+        vec!["input-parameters-dir/malformed-template.yaml"],
+        StatusCode::INTERNAL_FAILURE
+    )]
+    #[case(
+        vec!["db_resource.yaml"],
+        vec!["db_param_port_rule.guard"],
+        vec!["input-parameters-dir/blank-template.yaml"],
+        StatusCode::INTERNAL_FAILURE
+    )]
+    #[case(
+        vec!["db_resource.yaml"],
+        vec!["db_param_port_rule.guard"],
+        vec!["input-parameters-dir/blank-template.yaml", "input-parameters-dir/db_params.yaml"],
+        StatusCode::INTERNAL_FAILURE
+    )]
     fn test_combinations_of_rules_data_and_input_params_files(
         #[case] data_arg: Vec<&str>,
         #[case] rules_arg: Vec<&str>,
         #[case] input_params_arg: Vec<&str>,
         #[case] expected_status_code: i32,
     ) {
+        let mut reader = Reader::new(Stdin(std::io::stdin()));
         let mut writer = Writer::new(Stdout(stdout()), Stderr(stderr()));
         let status_code = ValidateTestRunner::default()
             .data(data_arg)
             .rules(rules_arg)
             .input_parameters(input_params_arg)
-            .run(&mut writer);
+            .run(&mut writer, &mut reader);
 
         assert_eq!(expected_status_code, status_code);
     }
+
+    #[test]
+    fn test_rules_with_data_from_stdin_yaml() {
+        let mut reader = get_reader(
+            "resources/validate/data-dir/s3-server-side-encryption-template-compliant.yaml",
+        );
+        let mut writer = Writer::new(Stdout(stdout()), WBVec(vec![]));
+        let status_code = ValidateTestRunner::default()
+            .rules(vec!["rules-dir/s3_bucket_public_read_prohibited.guard"])
+            .run(&mut writer, &mut reader);
+
+        assert_eq!(StatusCode::SUCCESS, status_code);
+    }
+
+    #[test]
+    fn test_rules_with_data_from_stdin_yaml_verbose() {
+        let mut reader = get_reader(
+            "resources/validate/data-dir/s3-server-side-encryption-template-compliant.yaml",
+        );
+        let mut writer = Writer::new(WBVec(vec![]), WBVec(vec![]));
+        let status_code = ValidateTestRunner::default()
+            .rules(vec!["rules-dir/s3_bucket_public_read_prohibited.guard"])
+            .verbose(true)
+            .run(&mut writer, &mut reader);
+
+        assert_output_from_file_eq!(
+            "resources/validate/output-dir/payload_verbose_success.out",
+            writer
+        );
+        assert_eq!(StatusCode::SUCCESS, status_code);
+    }
+
+    fn get_reader(path: &str) -> Reader {
```

Comment:
Should we put this method in `tests/utils.rs` as this may be used by integration tests for other sub-commands as well?

Reply:
great point